modThread/modThread.py

```diff
@@ -4,280 +4,404 @@
 
-    @commands.guild_only()
-    @commands.command()
-    @checks.admin_or_permissions(manage_guild=True)
-    async def unsetModsCategory(self, ctx):
-        """Unsets the mod category. Thread will not be moved if no mod category is set"""
-        await self._save_mods_category(ctx, None)
-        await ctx.send("Done")
+        await ctx.send(embed=settings_embed)
 
-    ### End Mod Category
+    @modthread.command(name="settings")
+    async def settings(
+        self,
+        ctx: commands.Context
+    ) -> None:
+        """Settings command group"""
 
-    # endregion
+        primary_category = await self._get_primary_category(ctx.guild)
+        management_role = await self._get_management_role(ctx.guild)
 
-    # region json db
-    async def _primary_category(self, ctx):
-        return ctx.guild.get_channel(
-            await self.config.guild(ctx.guild).PrimaryCategory()
+        settings_embed = discord.Embed(
+            title="ModThread Settings",
+            description="Current configuration for modThread Cog.",
+            color=discord.Color.blue(),
         )
 
-    async def _save_primary_category(self, ctx, primary_category):
-        await self.config.guild(ctx.guild).PrimaryCategory.set(primary_category)
-
-    async def _rules_category(self, ctx):
-        return ctx.guild.get_channel(await self.config.guild(ctx.guild).RulesCategory())
-
-    async def _save_rules_category(self, ctx, rules_category):
-        await self.config.guild(ctx.guild).RulesCategory.set(rules_category)
-
-    async def _numbers_category(self, ctx):
-        return ctx.guild.get_channel(
-            await self.config.guild(ctx.guild).NumbersCategory()
-        )
+        # Check channel values before mention to avoid exception
+        if primary_category:
+            settings_embed.add_field(
+                name="Primary Category",
+                value=primary_category.jump_url,
+                inline=False
+            )
+        else:
+            settings_embed.add_field(
+                name="Primary Category", value="Not Set", inline=False
+            )
 
-    async def _save_numbers_category(self, ctx, numbers_category):
-        await self.config.guild(ctx.guild).NumbersCategory.set(numbers_category)
+        if management_role:
+            settings_embed.add_field(
+                name="Management Role",
+                value=management_role.mention,
+                inline=False
+            )
+        else:
+            settings_embed.add_field(
+                name="Management Role", value="Not Set", inline=False
+            )
 
-    async def _mods_category(self, ctx):
-        return ctx.guild.get_channel(await self.config.guild(ctx.guild).ModsCategory())
+        await ctx.send(embed=settings_embed)
+
+    @modthread.command(name="groups", alias="group")
+    async def groups(
+        self,
+        ctx: commands.Context,
+        action: str | None,
```

Comment:
We should change this typing to a `Literal`. `Literal["delete", "rm", etc...]`

Because of discordpy converters, this will actually force you to use a correct action choice.

Reply:
I tested this out just now and hate it.

When I have this be `str | None`, this function will run and I can provide an error message to the user (invalid command).

When I use `Literal`, any value provided not in that Literal object is ignored and treated as None, so I can't correctly respond with an error message to the user and provide feedback/usage information.

Reply:
```python
    @modthread.command(name="groups", alias="group")
    async def groups(
        self,
        ctx: commands.Context,
        action: Literal["add", "update", "delete", "rm", "unset", "clear"] | None,
        group: str | None,
        category: discord.CategoryChannel | None,
        role: discord.Role | None
    ):
        """Groups command group"""

        if action:
            action = action.lower()
            if action in ["add", "update"]:
                group = group.lower()
                await self._set_group(ctx.guild, group, category, role)
                await ctx.send('Ok!')
            elif action in ['delete', 'rm', 'clear', 'unset']:
                group = group.lower()
                await self._unset_group(ctx.guild, group)
            else:
                await ctx.send(
                    'Unrecognized command. [`add`, `update`, `delete`]'
                )
```
![image](https://github.com/RSC-NA/bot-RSCBot/assets/124588/ed6dfd84-81a5-4123-af65-97e12eec659c)

This is the "tested" working code using your suggestion, and it takes away features that I like (custom error messaging)

![image](https://github.com/RSC-NA/bot-RSCBot/assets/124588/f7097630-3e6b-48ae-9fd3-dd0fe6d177e7)